Dev boro

MKurtz666 opened the pull request:
pliczek HangM2.py to hangman oparty o API datamuse. Prośba o review w wolnym czasie.

mcekiera commented on HangM2.py:
```diff
@@ -0,0 +1,87 @@
+        if ans == secretword[n]:                        # located in the secret word and to put it at analogous
+          guesses[n] = str(ans)                         # index in quesses: we want to show the player their
+          print("Yes! You got one. \n")                 # partial answer
+          partial = "".join([x for x in guesses])       # turning list into string for printing of partial answer
```
guesses jest już listą, nie ma potrzeby żeby używać tutaj comprahension, bo to w zasadzie przepisanie listy w listę, wystarczy tu sam join na liscie

mcekiera commented on HangM2.py:
```diff
@@ -0,0 +1,87 @@
+
+guessword()
+
+input("Press ENTER to quit.")
```
a to w sumie do czego? program zamiast sie po prostu zakoczyc, wymaga jeszcze wcisniecia entera

mcekiera commented on HangM2.py:
```diff
@@ -0,0 +1,87 @@
+
+def secret_word_selection():
+  parameter = {noun_or_adj():topic_select(), "max":100}              # creating the full parameter for the api end point
+  req = (requests.get("http://api.datamuse.com/words",parameter)).json()      #reaching data in end point
```
co do użycia API to elegancko poszło widze :+1:

mcekiera commented on HangM2.py:
```diff
@@ -0,0 +1,87 @@
+def topic_select():                # input that will determine the keyword responsible for topic association
+  while True:
+    topicsel = (str(input("Enter a topic a word related to which you want to guess: \n"))).lower()
+    if any(char.isdigit() for char in topicsel):    # safeguard against integers in input
```
:+1:

mcekiera commented on HangM2.py:
```diff
@@ -0,0 +1,87 @@
+input("Press ENTER to quit.")
+
+
+
```
dwie kwestie bym przemyślał, czy nie da się wydzielić mniejszych funkcji :) a druga sprawa, czy nie dało by się tego napisać tak, żeby warstwa modelu danych (słowo, zgadywanie literek, etc) była odzielona od widoku (w tej wersji od printów do konsoli), np. w taki sposób obracać danymi o zgadwanym słowie, żeby Ci było wszystko jedno jak to byłoby wyświetlane, ale to nie jest prosta sprawa wiec na luzie ;P

MKurtz666 commented:
Elo, 

Skomitowałem programik Herzog_Dungeon do brancha develop i dałem komentarz do commita.

Jakbyś miał chwilęto review bardzo mile widziane 😊 No i zagraj ofkors bo to gra w końcu.

Pzdr,
MK


Wysłane z aplikacji Poczta dla Windows 10

Od: Marcin Cekiera
Wysłano: środa, 4 marca 2020 22:15
Do: MKurtz666/CentrumKolaboracji
DW: MKurtz666; Author
Temat: Re: [MKurtz666/CentrumKolaboracji] Dev boro (#5)

@mcekiera commented on this pull request.

In HangM2.py:
> @@ -0,0 +1,87 @@

+import requests

+import random

+

+def noun_or_adj():

+  while True:

+    selection = input("Enter A for an adjective or N for a noun: \n\n")     # input that will determine the keyword

+    if selection == ("a").lower():                                           # for the API

+      return "rel_jjb"             # per the API documentation this means 'select only adjectives'

+      break

return przerywa wykonanie funkcji i zwraca daną wartość, wiec ten break nie jest tutaj potrzebny, break ma bardziej zastosowanie, kiedy chcesz przerwać wykonywanie pętli, bez zwracania wartości

In HangM2.py:
> @@ -0,0 +1,87 @@

+import requests

+import random

+

+def noun_or_adj():

+  while True:

+    selection = input("Enter A for an adjective or N for a noun: \n\n")     # input that will determine the keyword

+    if selection == ("a").lower():                                           # for the API

+      return "rel_jjb"             # per the API documentation this means 'select only adjectives'

+      break

+    elif selection == ("n").lower():

+      return "rel_jja"             # per the API documentation this means 'select only nouns'

+      break

+    else:

+      print("Sorry, wrong input. Please try again: \n\n")

+      continue                     # safeguard for wrong input

continue też nie jest potrzebny, jeżeli jest ostatnim poleceniem w iteracji, bez tego i tak rozpoczenie się kolejna, używa się tego raczej, jeżeli zaistniał warunek, by nie wykonywać czesci kodu która znajduje sie po continue

In HangM2.py:
> @@ -0,0 +1,87 @@

+import requests

+import random

+

+def noun_or_adj():

+  while True:

+    selection = input("Enter A for an adjective or N for a noun: \n\n")     # input that will determine the keyword

+    if selection == ("a").lower():                                           # for the API

+      return "rel_jjb"             # per the API documentation this means 'select only adjectives'

zamiast komentarza z wytłumaczeniem, lepsza byłaby zmienna o czytelnej nazwie, np. SELECT_ONLY_ADJ, etc, która by miałą wartość 'rel_jjb', to samo poniżej

In HangM2.py:
> +    if selection == ("a").lower():                                           # for the API

+      return "rel_jjb"             # per the API documentation this means 'select only adjectives'

+      break

+    elif selection == ("n").lower():

+      return "rel_jja"             # per the API documentation this means 'select only nouns'

+      break

+    else:

+      print("Sorry, wrong input. Please try again: \n\n")

+      continue                     # safeguard for wrong input

+

+def topic_select():                # input that will determine the keyword responsible for topic association

+  while True:

+    topicsel = (str(input("Enter a topic a word related to which you want to guess: \n"))).lower()

+    if any(char.isdigit() for char in topicsel):    # safeguard against integers in input

+      print("Sorry, wrong input.\n\n")

+      continue

tutaj tak naprawde continue też jest ostatnia wykonywana komenda, bo nie ma nic poza struktura if/elif/else

In HangM2.py:
> +    if any(char.isdigit() for char in topicsel):    # safeguard against integers in input

+      print("Sorry, wrong input.\n\n")

+      continue

+    elif len(topicsel.split()) > 1:                 # safeguard against multiple words in input

+      print("Sorry, wrong input. Please limit answer to a single word.\n\n")

+    else:

+      return str(topicsel)

+

+def secret_word_selection():

+  parameter = {noun_or_adj():topic_select(), "max":100}              # creating the full parameter for the api end point

+  req = (requests.get("http://api.datamuse.com/words",parameter)).json()      #reaching data in end point

+  words = []

+  for w in req:                                                               # creating an array of possible words

+    words.append(w["word"])                                                   # that meet the key criteria

+  secretword = random.choice(words)                                           # selecting one of those words

+  return str(secretword)

czy tu faktycznie jest potrzeba by rzutować to do stringa? co tam jest zwracane że musisz to robić?

In HangM2.py:
> +      continue

+    elif len(topicsel.split()) > 1:                 # safeguard against multiple words in input

+      print("Sorry, wrong input. Please limit answer to a single word.\n\n")

+    else:

+      return str(topicsel)

+

+def secret_word_selection():

+  parameter = {noun_or_adj():topic_select(), "max":100}              # creating the full parameter for the api end point

+  req = (requests.get("http://api.datamuse.com/words",parameter)).json()      #reaching data in end point

+  words = []

+  for w in req:                                                               # creating an array of possible words

+    words.append(w["word"])                                                   # that meet the key criteria

+  secretword = random.choice(words)                                           # selecting one of those words

+  return str(secretword)

+

+secretword = secret_word_selection()

a to możesz wywołać już w funkcji guessword, możesz łatwo rozszerzyć gierkę o funkcjonalność 'zagraj ponownie' :P

In HangM2.py:
> +    words.append(w["word"])                                                   # that meet the key criteria

+  secretword = random.choice(words)                                           # selecting one of those words

+  return str(secretword)

+

+secretword = secret_word_selection()

+

+print("The secret word has " + str(len(secretword)) + " letters. \n\n")     # a clue for the player

+

+def guessword():

+  global secretword

+

+  # defining variables:

+

+  counter = 0                           # counter to help control the number of iterations

+  max_counter = len(secretword)         # the limit of iterations based on counter

+  guesses = list(len(secretword)*"_")   # create a list - empty at first will be populated with guessed letters

kosmetyka, ale masz już wartość len(secretword) w zmiennej max_counter, możesz ją użyć, no i warto dla czytleności kodu dodawać po spacji między znakami działań matematycznych (+-*/)

In HangM2.py:
> +print("The secret word has " + str(len(secretword)) + " letters. \n\n")     # a clue for the player

+

+def guessword():

+  global secretword

+

+  # defining variables:

+

+  counter = 0                           # counter to help control the number of iterations

+  max_counter = len(secretword)         # the limit of iterations based on counter

+  guesses = list(len(secretword)*"_")   # create a list - empty at first will be populated with guessed letters

+

+  # commencing loop:

+  while True:

+

+    ans = input("""Enter your quess (letter or word)

+or enter '0' if you\'re a little quitter bitch: \n\n""")  # user input

🤣

In HangM2.py:
> +

+def guessword():

+  global secretword

+

+  # defining variables:

+

+  counter = 0                           # counter to help control the number of iterations

+  max_counter = len(secretword)         # the limit of iterations based on counter

+  guesses = list(len(secretword)*"_")   # create a list - empty at first will be populated with guessed letters

+

+  # commencing loop:

+  while True:

+

+    ans = input("""Enter your quess (letter or word)

+or enter '0' if you\'re a little quitter bitch: \n\n""")  # user input

+    if ans == "0":

to moze wydawać się overkill, ale jakbyś utworzył zmienną np. EXIT o wartości "0" lub sprawdzał to np. w ramach funkcji isUserLeaving(input), to na pierwszy rzut oka ktoś by wiedział jaki warunek tu rozpatrujesz

In HangM2.py:
> +      continue                     # safeguard for wrong input

+

+def topic_select():                # input that will determine the keyword responsible for topic association

+  while True:

+    topicsel = (str(input("Enter a topic a word related to which you want to guess: \n"))).lower()

+    if any(char.isdigit() for char in topicsel):    # safeguard against integers in input

+      print("Sorry, wrong input.\n\n")

+      continue

+    elif len(topicsel.split()) > 1:                 # safeguard against multiple words in input

+      print("Sorry, wrong input. Please limit answer to a single word.\n\n")

+    else:

+      return str(topicsel)

+

+def secret_word_selection():

+  parameter = {noun_or_adj():topic_select(), "max":100}              # creating the full parameter for the api end point

+  req = (requests.get("http://api.datamuse.com/words",parameter)).json()      #reaching data in end point

a propos nazwy req, ta zmienna zawiera bardziej response niz request

In HangM2.py:
> @@ -0,0 +1,87 @@

+import requests

+import random

+

+def noun_or_adj():

zaczynasz rozbijać całość na funkcje, imo to krok w dobra strone ;) ale wydaje mi się że możesz tutaj więcej mniejszych funkcji wydzielić, szczególnie że tworzsz kilka wersji programu, więc spokojnie możesz wydzielić funkję i użyć jej w kilku propgramach ;) wiesz jak działa import, wiec możesz też importować swoje funkcje

In HangM2.py:
> +      print("The secret word was: " + secretword)

+      break

+    elif len(ans) > 1:                                  # safeguard against partial word input since we want the user

+      if ans == secretword:                             # to input either a single char or the entire word

+        print(secretword)

+        print("You win.\n")

+        break

+      else:

+        print("Nope, that\'s not it.\n")

+        counter += 1

+    elif ans in secretword:                             # if answer char is indeed a part of the secret word

+      for n in range(len(secretword)):                  # a quick loop to determine the index at which the char is

+        if ans == secretword[n]:                        # located in the secret word and to put it at analogous

+          guesses[n] = str(ans)                         # index in quesses: we want to show the player their

+          print("Yes! You got one. \n")                 # partial answer

+          partial = "".join([x for x in guesses])       # turning list into string for printing of partial answer

guesses jest już listą, nie ma potrzeby żeby używać tutaj comprahension, bo to w zasadzie przepisanie listy w listę, wystarczy tu sam join na liscie

In HangM2.py:
> +          guesses[n] = str(ans)                         # index in quesses: we want to show the player their

+          print("Yes! You got one. \n")                 # partial answer

+          partial = "".join([x for x in guesses])       # turning list into string for printing of partial answer

+          print(partial)

+    else:

+      counter += 1                                      # if answer is neither the secret word nor a part of it

+      print("Nope, try again. \n")                      # counter + 1 and message

+      partial = "".join([x for x in guesses])           # turning list into string for printing of partial answer

+      print(partial)

+    if counter >= max_counter:                          # loop termination if counter reaches limit

+      print("No more tries. Better luck next time.\n")

+      break

+

+guessword()

+

+input("Press ENTER to quit.")

a to w sumie do czego? program zamiast sie po prostu zakoczyc, wymaga jeszcze wcisniecia entera

In HangM2.py:
> +      continue                     # safeguard for wrong input

+

+def topic_select():                # input that will determine the keyword responsible for topic association

+  while True:

+    topicsel = (str(input("Enter a topic a word related to which you want to guess: \n"))).lower()

+    if any(char.isdigit() for char in topicsel):    # safeguard against integers in input

+      print("Sorry, wrong input.\n\n")

+      continue

+    elif len(topicsel.split()) > 1:                 # safeguard against multiple words in input

+      print("Sorry, wrong input. Please limit answer to a single word.\n\n")

+    else:

+      return str(topicsel)

+

+def secret_word_selection():

+  parameter = {noun_or_adj():topic_select(), "max":100}              # creating the full parameter for the api end point

+  req = (requests.get("http://api.datamuse.com/words",parameter)).json()      #reaching data in end point

co do użycia API to elegancko poszło widze 👍

In HangM2.py:
> +  while True:

+    selection = input("Enter A for an adjective or N for a noun: \n\n")     # input that will determine the keyword

+    if selection == ("a").lower():                                           # for the API

+      return "rel_jjb"             # per the API documentation this means 'select only adjectives'

+      break

+    elif selection == ("n").lower():

+      return "rel_jja"             # per the API documentation this means 'select only nouns'

+      break

+    else:

+      print("Sorry, wrong input. Please try again: \n\n")

+      continue                     # safeguard for wrong input

+

+def topic_select():                # input that will determine the keyword responsible for topic association

+  while True:

+    topicsel = (str(input("Enter a topic a word related to which you want to guess: \n"))).lower()

+    if any(char.isdigit() for char in topicsel):    # safeguard against integers in input

👍

In HangM2.py:
> +          print(partial)

+    else:

+      counter += 1                                      # if answer is neither the secret word nor a part of it

+      print("Nope, try again. \n")                      # counter + 1 and message

+      partial = "".join([x for x in guesses])           # turning list into string for printing of partial answer

+      print(partial)

+    if counter >= max_counter:                          # loop termination if counter reaches limit

+      print("No more tries. Better luck next time.\n")

+      break

+

+guessword()

+

+input("Press ENTER to quit.")

+

+

+

dwie kwestie bym przemyślał, czy nie da się wydzielić mniejszych funkcji :) a druga sprawa, czy nie dało by się tego napisać tak, żeby warstwa modelu danych (słowo, zgadywanie literek, etc) była odzielona od widoku (w tej wersji od printów do konsoli), np. w taki sposób obracać danymi o zgadwanym słowie, żeby Ci było wszystko jedno jak to byłoby wyświetlane, ale to nie jest prosta sprawa wiec na luzie ;P
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.



-- 
Ta wiadomość została sprawdzona na obecność wirusów przez oprogramowanie antywirusowe Avast.
https://www.avast.com/antivirus
